Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/NonprofitSignupViewController.m

```diff
@@ -18,28 +19,58 @@ @interface NonprofitSignupViewController ()
 @property (weak, nonatomic) IBOutlet UITextField *nonprofitCategoryTextField;
 @property (weak, nonatomic) IBOutlet UITextField *nonprofitWebsiteTextField;
 //TODO: set up with Paypal, Braintree?
-
+@property (nonatomic) BOOL finishedSavingBoth;
 
 @end
 
 @implementation NonprofitSignupViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    self.finishedSavingBoth = false;
     self.nonprofit = [Nonprofit new]; //initWithDict
     //TODO: figure out document uplaod
     // https://stackoverflow.com/questions/37296929/implement-document-picker-in-swift-ios
 }
 
 
 - (IBAction)getStartedButtonTapped:(id)sender {
-    //nonprofit property is still nil
-    self.user.nonprofit = self.nonprofit;
-    //save user, save nonprofit
-    
-    [self performSegueWithIdentifier:@"loginSegue" sender:nil];
+    [self saveNonprofitAndUserToParse];
 }
 
+-(void)saveNonprofitAndUserToParse {
+    self.nonprofit.profilePicFile;
+    self.nonprofit.nonprofitName = self.nonprofitNameTextField.text;
+    self.nonprofit.nonprofitDescription = self.nonprofitDescriptionTextView.text;
+    self.nonprofit.category = self.nonprofitCategoryTextField.text;
+    self.nonprofit.websiteUrlString = [NSString stringWithFormat:@"%@", self.nonprofitWebsiteTextField.text];
+    self.user.nonprofit = self.nonprofit;
+    [self.user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+        if (succeeded) {
+            if (self.finishedSavingBoth) {
```

Comment:
This synchronization logic seems a bit fragile and prone to race conditions, I would recommend using a `NSOperationQueue` with a `completionBlock` instead, or just serializing the two queries as they're probably very fast anyway.